### PR TITLE
options.md

### PR DESCRIPTION
**Pull request by MarioRicalde**
A draft with what you can set to customize, also ensuring that the one-liners that are fetched from the file are properly written. 

- [x] Draft.
- [ ] Document all variables. 
- [ ] Previews.

### CODE REVIEW

**Comment by MarioRicalde** [comment]
@jedahan @justmendes can you guys double check to make sure I'm not missing any variable?

Also, I'm thinking once we have this file we need to copy it and just add the older variable equivalents, and then we can do the automation and the updating guide. 

Thoughts?

**Comment by jedahan** [comment]
My main suggestion is to find a formatting that makes it a bit easier to scan - all the backticks visually feel cramped. Perhaps using tables, especially for git, perhaps something like:

##  [geometry_git.zsh](functions/geometry_git.zsh)

|`geometry_git_stashes()`|  |
|:-----------------------|--------|
|GEOMETRY_GIT_COLOR_STASHES | 144|
|GEOMETRY_GIT_SYMBOL_STASHES | ●|

|`geometry_git_time()`|  |
|:-----------------------|--------|
GEOMETRY_COLOR_NO_TIME | white
GEOMETRY_GIT_NO_COMMITS_MESSAGE | no-commits 
GEOMETRY_GIT_TIME_DETAILED|false

**Comment by MarioRicalde** [comment]
Most definitely, I'm going to be adding proper formatting to this document. What you see at the moment is just a list of variables, definitely not ready.

Is there anything in there that you notice I'm missing? If not, I can get to finishing this today.

**Comment by frm** [comment]
Nice work!

Some minor things: can we put this into a `docs` directory? Also, can you add a title (e.g `# Customization Options`) and a small intro paragraph explaining how to do that (adding to zshrc, before sourcing geometry, etc)?

File: options.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+`GEOMETRY_STATUS_SYMBOL_ROOT_ERROR` - ▽
+`GEOMETRY_STATUS_SYMBOL_COLOR_HASH` - false
+
+### `geometry_virtualenv.zsh`
+
+Show the current `virtualenv` or `conda` environment.
+
+`GEOMETRY_VIRTUALENV_CONDA_SEPARATOR` - :
+`GEOMETRY_VIRTUALENV_COLOR` - `DEFAULT_COLOR`
+`GEOMETRY_VIRUALENV_CONDA_COLOR` - `DEFAULT_COLOR`
```
Comment (jedahan) [inline options.md]: now both of these are simply 'green' and easier to override

File: options.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+`GEOMETRY_RUBY_SYMBOL` - ◆
+`GEOMETRY_RUBY_RVM_SHOW_GEMSET` - true
+
+### `geometry_rust_version.zsh`
+
+Display the current version of rust (by @drager).
+
+`GEOMETRY_RUST_VERSION_COLOR` - red
+
+### `geometry_rustup.zsh`
```
Comment (jedahan) [inline options.md]: there is GEOMETRY_RUSTUP_PIN can be set to keep rustup rendering even out of context

`also GEOMETRY_RUSTUP_SYMBOL:=⚙`

File: options.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+`GEOMETRY_JOBS_SYMBOL` - ⚙
+
+### `geometry_kube.zsh`
+
+Show kubectl (Kubernetes) client version and current context/namespace.
+
+`GEOMETRY_KUBE_COLOR` - blue
+`GEOMETRY_KUBE_SYMBOL` - ⎈
+
+### `geometry_node.zsh`
```
Comment (jedahan) [inline options.md]: GEOMETRY_NODE_PIN can be set to always show geometry_node, even outside the context of a node project folder

File: options.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+`GEOMETRY_HOSTNAME_SEPARATOR` - @
+
+### `geometry_jobs.zsh`
+
+Shows background jobs.
+
+`GEOMETRY_JOBS_COLOR` - blue
+`GEOMETRY_JOBS_SYMBOL` - ⚙
+
+### `geometry_kube.zsh`
```
Comment (jedahan) [inline options.md]: GEOMETRY_KUBE_PIN as below

File: options.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+### `geometry_exec_time.zsh`
+
+Show the elapsed time for long running commands.
+
+Options:
+`GEOMETRY_EXEC_TIME_FILE` - Path to temp directory.
+`GEOMETRY_EXEC_TIME_PATIENCE` - Seconds before the time is shown.
+
+### `geometry_git.zsh`
```
Comment (jedahan) [inline options.md]: GEOMETRY_GIT_GREP (though its in conflicts, I think it should be top level) can be used to override our searching for rg > ag > grep

File: options.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
+### `geometry_docker_machine.zsh`
+
+Show the docker machine name.
+
+Options: none.
```
Comment (jedahan) [inline options.md]: `${GEOMETRY_DOCKER_MACHINE_COLOR:=blue} "(${GEOMETRY_DOCKER_MACHINE_SYMBOL:="⚓"}`

**Review by jedahan — CHANGES_REQUESTED** [review]
There are a decent number of inline GEOMETRY_ env options, not all are defined with `: ${GEOMETRY_THING...` syntax.

File: options.md (hunk 1 above)
Comment (MarioRicalde) [reply options.md]: Great job. I was wondering what was the reasoning behind keeping it like that!

File: options.md [hunk 7]
```diff
@@ -128,6 +133,8 @@ Display a symbol colored with the currently selected rustup toolchain.
 `GEOMETRY_RUSTUP_STABLE_COLOR` - green
 `GEOMETRY_RUSTUP_BETA_COLOR` - yellow
 `GEOMETRY_RUSTUP_NIGHTLY_COLOR` - red
+`GEOMETRY_RUSTUP_PIN` - Can be setup to keep rustup rendering even out of context.
+``GEOMETRY_RUSTUP_SYMBOL` - ⚙
```
Comment (jedahan) [inline options.md]: accidental double backtick?

**Comment by MarioRicalde** [comment]
@jedahan I added the variables you suggested. 👍 

@justmendes I've pushed the introduction.

I'm still pending formatting.

File: options.md [hunk 8]
```diff
@@ -1,3 +1,8 @@
+# Customization Options
+
+Configuring Geometry is just a matter of setting some variables **before** sourcing the theme itself. Below, you'll find a list of all possible options that you can set that will override Geometry appearance and functionality.
```
Comment (jedahan) [inline options.md]: So I think you don't need to set the variables before anymore. Most functions read the variables in live. If you find any that don't, we should file bugs.

Comment (MarioRicalde) [reply options.md]: That's good to know, updating.

File: options.md (hunk 7 above)
Comment (MarioRicalde) [reply options.md]: Yep, resolving. I'll work on formatting next.

**Comment by MarioRicalde** [comment]
I've added some table formatting to the document; however, there's still several that are pending documentation in the description.

One thing that I'm thinking on doing is taking screenshots of the prompts to go hand in hand with these tables, but maybe down the line.

**Comment by MarioRicalde** [comment]
@jedahan I've added descriptions to the best of my ability - but I'm sure I'm missing something. Also, the table of contents is there. Could you please review and advise if there's anything that needs to change.

Thank you.

File: options.md [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+Configuring Geometry is just a matter of setting some variables. Below, you'll find a 
+list of all possible options that you can set that will override Geometry appearance 
+and functionality.
+
+## Table of Contents
+
+- [Customization Options](#customization-options)
+  - [Table of Contents](#table-of-contents)
+  - [`geometry_docker_machine.zsh`](#geometrydockermachinezsh)
+      - [Options](#options)
```
Comment (jedahan) [inline options.md:12]: I would remove the options ToC link for everything - even with git we can just have it like follows:
```markdown
- [`geometry_git.zsh`](#geometrygitzsh)
  - [`geometry_git_stashes()`](#geometrygitstashes)
```

Comment (jedahan) [reply options.md:12]: Actually how about removing the `####options` subheader from all sections as it feels redundant?

**Review by jedahan — APPROVED** [review]
Minor nitpick, but I am happy to merge this in, thanks for all the hard work!

**Pull request merged**